jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt

```diff
@@ -158,6 +159,12 @@ class CodeWhispererExplorerActionManager : PersistentStateComponent<CodeWhispere
         actionState.accountlessErrorTimestamp = state.accountlessErrorTimestamp
     }
 
+    private fun isConnectionExpired(project: Project): Boolean {
+        val tokenProvider = CodeWhispererUtil.tokenProvider(project) ?: return true
```

Comment:
In what condition would tokenProvider be null, again?

Reply:
logically, shouldn't be the case, however since function `tokenProvider` use `as?` to safely cast type therefore the null case.